runbotics/runbotics-scheduler/src/scheduler-database/tenant/tenant.service.ts

```diff
@@ -0,0 +1,93 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { MoreThanOrEqual, Repository } from 'typeorm';
+
+import { UserEntity } from '#/database/user/user.entity';
+import { Tenant } from './tenant.entity';
+import { CreateTenantDto } from './dto/create-tenant.dto';
+import { UpdateTenantDto } from './dto/update-tenant.dto';
+import { Logger } from '#/utils/logger';
+import { TenantInviteCode } from './tenant-invite-code.entity';
+import dayjs from 'dayjs';
+import { TenantInviteCodeDto } from './dto/invite-code.dto';
+
+@Injectable()
+export class TenantService {
+    private readonly logger = new Logger(TenantService.name);
+
+    constructor(
+        @InjectRepository(Tenant)
+        private readonly tenantRepository: Repository<Tenant>,
+        @InjectRepository(TenantInviteCode)
+        private readonly inviteCodeRepository: Repository<TenantInviteCode>
+    ) {}
+
+    getTenantById(id: string) {
+        return this.tenantRepository.findOneBy({ id });
+    }
+
+    getAllTenants() {
+        return this.tenantRepository.find();
+    }
+
+    async createTenant(tenantDto: CreateTenantDto, requester: UserEntity): Promise<Tenant> {
+        const tenantByName = await this.tenantRepository.findOneBy({ name: tenantDto.name });
+        if (tenantByName) throw new BadRequestException('Name already exist', 'NameExist');
+
+        const newTenant = {
+            ...tenantDto,
+            created: dayjs(),
+            updated: dayjs(),
+            createdById: requester.id,
+            lastModifiedBy: requester.email
+        };
+
+        return this.tenantRepository.save(newTenant);
+    }
+
+    async updateTenant(
+        tenantDto: UpdateTenantDto,
+        id: string,
+        requester: UserEntity
+    ): Promise<Tenant> {
+        const tenantByName = await this.tenantRepository.findOneBy({ name: tenantDto.name });
+        if (tenantByName) throw new BadRequestException('Name already exist', 'NameExist');
+
+        const updatedTenant = await this.tenantRepository
+            .findOneByOrFail({ id })
+            .then(tenant => ({
+                ...tenant,
+                ...tenantDto,
+                updated: dayjs(),
+                lastModifiedBy: requester.email
+            })).catch(() => {
+                this.logger.error('Cannot find tenant with id: ', id);
+                throw new BadRequestException('Tenant not found', 'NotFound');
+            });
```

Comment:
i think even if tenant if found but in "then" expression while updatating the error occurs - the "catch" is going to be executed, so for example if you try to update **existing tenant** that is found but with e.g. FK or property that does not exist (wchich throws the error) the misleading error is going to be returned - tenant not found